docs/advanced_usage.md

```diff
@@ -0,0 +1,232 @@
+subsection.
+
+### Rotate keys
+
+In case of potential compromise, notary provides a CLI command for rotating keys. Currently, you can use the `notary key rotate` command to rotate the targets or snapshot keys.
+
+While the snapshot key is managed by the notary client by default, use the `notary key
+rotate -r` command to rotate the snapshot key to the server, such that the
+notary server will then sign snapshots. This is particularly useful when using
+delegations with a repository, so that delegates will never need access to the
+snapshot key to push their updates to the repository.
+
+### Use a Yubikey
+
+Notary can be used with [Yubikey
+4](https://www.yubico.com/products/yubikey-hardware/yubikey4/) keys, via the
+PKCS11 interface with CCID-enabled PIV. The Yubikey will be prioritized to store
+root keys, and will require user touch-input for signing.
+
+>**Note**: Yubikey support for signing docker images is only supported in the experimental branch.
+
+Yubikey support requires [Yubico PIV libraries (which are bundled with the PIV
+tools)](https://www.yubico.com/support/downloads) to be available in standard
+library locations.
+
+## Work with delegation roles
+
+Delegation roles simplify collaborator workflows in notary repositories, and
+also allow for fine-grained permissions within a repository's contents across
+delegations. In essence, delegation roles are restricted versions of the targets
+role that are only allowed to sign targets within certain filepaths.
+
+A delegation role is given its own keys, such that each collaborator can keep
+his own private key without the repository administrator having to share the
+targets key or allow a collaborator write access to all targets of the
+repository.
+
+Before adding any delegations, you should rotate the snapshot key to the server.
+This is such that delegation roles will not require the snapshot key to publish
+their own targets to the repository, since the server can publish the valid
+snapshot with the delegation targets:
+
+```
+$ notary key rotate example.com/collection -r --key-type=snapshot
+```
+
+When adding a delegation, your must acquire a x509 certificate with the public
+key of the user you wish to delegate to. The user who will assume this
+delegation role must hold the private key to sign content with notary.
+
+Once you've acquired the delegate's x509 certificate, you can add a delegation
+for this user:
+
+```
+$ notary delegation add example.com/collection targets/releases cert.pem --paths="delegation/path"
+```
+
+In the preceding example, illustrates a request to add the delegation
+`targets/releases` to the GUN `example.com/collection`. The delegation name must
+be prefixed by `targets/` to be valid, since all delegations are restricted
+versions of the target role. The command adds the public key contained in the
+x509 cert `cert.pem` to the `targets/releases` delegation.  
+
+For the `targets/releases` delegation role to sign content, the delegation user
```

Comment:
We should be more explicit that this is a straight path prefix, maybe give a couple of examples:

```The provided paths are interpreted as simple prefixes on any target paths added to the role. With the given path of "delegation/path", the targets/releases role would be able to sign paths like "delegation/path/content.txt", "delegation/path_file.txt" and "delegation/path.txt"```